test/extended/openstack/loadbalancers.go

```diff
@@ -630,3 +746,39 @@ func getMembersMatchingStatus(client *gophercloud.ServiceClient, pool *pools.Poo
 	}
 	return members, nil
 }
+
+// Active wait until allowed_cidrs is updated in a given load balancer listener
+func waitUntilLbListenerAllowedCidrsIsUpdated(client *gophercloud.ServiceClient, interval time.Duration, timeout time.Duration, listenerId string, expectedCidrs []string) error {
```

Comment:
This is new for me but I found something that can make this poll stuff: https://onsi.github.io/gomega/#making-asynchronous-assertions. Not sure if that can be applicable here, maybe something like:

`o.Eventually(60, getAllowedCIDRs(client, listernerId)).Should(o.Consistof(AllowedCIDRs))`

Reply:
Nice catch! I replaced it and it works very well, I think Gomega Eventually and Consistently are really useful here :)